allauth/account/forms.py

```diff
@@ -556,6 +556,14 @@ def __init__(self, *args, **kwargs):
 
     def save(self):
         get_adapter().set_password(self.user, self.cleaned_data["password1"])
+        try:
+            # verify user's EmailAddress on password reset via email
+            email_address = EmailAddress.objects.get_for_user(self.user, user_email(self.user))
```

Comment:
A user can have multiple email addresses. Here, you happen to return the one that is one user.email, but it is not 100% certain that this was the email address the password reset mail was sent to.